apps/web/src/features/run-preview/hooks/use-preview-canvas-viewport-controller.ts

```
@@ -0,0 +1,124 @@
+    if (syncInFlightRef.current) {
+      syncQueuedRef.current = true;
+      return;
+    }
+
+    syncInFlightRef.current = true;
+    void (async () => {
+      try {
+        await runCanvasViewportSync();
+      } finally {
+        syncInFlightRef.current = false;
+        if (!syncQueuedRef.current) return;
```

Biome (2.5.1)

[error] Unsafe usage of 'return'.

(lint/correctness/noUnsafeFinally)